### PR TITLE
Use Dagger AndroidInjector in example app

### LINKED ISSUE #495: Bump Dagger to version 2.11 and simplify injection via `dagger.android` utils

### PR DESCRIPTION
**Pull request by aforcier**
Introduces [`dagger-android`](https://google.github.io/dagger/android.html) to the example app, and uses `AndroidInjector` to improve injection for the UI elements. Closes #495.

#586, which increases the Dagger version to the required `2.11`, should be reviewed and merged first.

cc @maxme

### CODE REVIEW

**Comment by aforcier** [comment]
Rebased out the previous reliance on the new kapt (#580), and fixed an issue with app secrets not being loaded. Ready for review @maxme!

File: example/src/main/java/org/wordpress/android/fluxc/example/CommentsFragment.java [hunk 1]
```diff
@@ -29,15 +30,17 @@
...
     @Inject SiteStore mSiteStore;
     @Inject CommentStore mCommentStore;
     @Inject Dispatcher mDispatcher;
 
     @Override
-    public void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        ((ExampleApp) getActivity().getApplication()).getComponent().inject(this);
+    public void onAttach(Context context) {
+        AndroidInjection.inject(this);
```
Comment (maxme) [inline example/src/main/java/org/wordpress/android/fluxc/example/CommentsFragment.java]: ❤️

**Comment by maxme** [comment]
Looks good, I'm fine to not update Instaflux to show how we can inject components using the old method.

**Comment by maxme** [comment]
:shipit:

**Pull request merged**